src/svg_surface.rs

```diff
@@ -0,0 +1,167 @@
+	}
+}
+
+impl AsRef<Surface> for SVGSurface {
+    fn as_ref(&self) -> &Surface {
+        &self.0
+    }
+}
+
+impl Deref for SVGSurface {
+    type Target = Surface;
+
+    fn deref(&self) -> &Surface {
+        &self.0
+    }
+}
+
+impl Clone for SVGSurface {
+    fn clone(&self) -> SVGSurface {
+        SVGSurface(self.0.clone())
+    }
+}
+
+unsafe impl Send for SVGSurface {}
+
+#[cfg(feature = "use_glib")]
+impl<'a> ToGlibPtr<'a, *mut ffi::cairo_surface_t> for SVGSurface {
+    type Storage = &'a Surface;
+
+    #[inline]
+    fn to_glib_none(&'a self) -> Stash<'a, *mut ffi::cairo_surface_t, Self> {
+        let stash = self.0.to_glib_none();
+        Stash(stash.0, stash.1)
+    }
+}
+
+#[cfg(feature = "use_glib")]
+impl FromGlibPtrNone<*mut ffi::cairo_surface_t> for SVGSurface {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut ffi::cairo_surface_t) -> SVGSurface {
+        Self::from(from_glib_none(ptr)).unwrap()
+    }
+}
+
+#[cfg(feature = "use_glib")]
+impl FromGlibPtrBorrow<*mut ffi::cairo_surface_t> for SVGSurface {
+    #[inline]
+    unsafe fn from_glib_borrow(ptr: *mut ffi::cairo_surface_t) -> SVGSurface {
+        Self::from(from_glib_borrow(ptr)).unwrap()
+    }
+}
+
+#[cfg(feature = "use_glib")]
+impl FromGlibPtrFull<*mut ffi::cairo_surface_t> for SVGSurface {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut ffi::cairo_surface_t) -> SVGSurface {
+        Self::from_raw_full(ptr)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use context::*;
+    use std::path::PathBuf;
+
+    fn draw_x(cr: &Context) {
+        cr.set_line_width(25.0);
+
+        cr.set_source_rgba(1.0, 0.0, 0.0, 0.5);
+        cr.line_to(0.,0.);
+        cr.line_to(100.,100.);
+        cr.stroke();
+
+        cr.set_source_rgba(0.0, 0.0, 1.0, 0.5);
+        cr.line_to(0.,100.);
+        cr.line_to(100.,0.);
+        cr.stroke();
+    }
+
+    #[test]
+    fn svg() {
+        let output = PathBuf::from( env!("OUT_DIR") );
+
+        { //Cow<str>
+            let filename = output.join("test1.svg");
+            let surface = SVGSurface::create(filename, 100., 100.);
+            let cr = Context::new(&surface);
+            draw_x(&cr);
+        }
+        { // &str
+            let filename = output.join("test2.svg");
+            let s = match filename.to_str() {
+                None => panic!("Error converting Path to String"),
+                Some(x) => x,
+            };
+            let surface = SVGSurface::create(s, 100., 100.);
+            let cr = Context::new(&surface);
+            draw_x(&cr);
+        }
+        { // String
+            let filename = output.join("test3.svg");
+            let s = match filename.to_str() {
+                None => panic!("Error converting Path to String"),
+                Some(x) => String::from(x),
+            };
+            let surface = SVGSurface::create(s, 100., 100.);
+            let cr = Context::new(&surface);
+            draw_x(&cr);
+        }
```

Comment:
@sdroege You don't think that many cases in this test actually try to test
rustc ability to generate call code? IMHO we need only one test without units and one with.

Reply:
I agree. But what could make sense is to add tests that actually check that the path conversion works correctly and the resulting files actually exist with their correct names.

Create some files with non-ASCII names and check with the API from `std` if those files actually exist with those names in the end. That would also solve my concern about the conversion as we run tests on Linux and Windows, and those are the platforms with the main differences.